library/Icingadb/ProvidedHook/CreateHostSlaReport.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Icinga\Module\Icingadb\ProvidedHook;
+
+use Icinga\Module\Icingadb\Hook\HostsDetailExtensionHook;
+use ipl\Html\Html;
+use ipl\Html\ValidHtml;
+use ipl\Orm\Query;
+use ipl\Web\Filter\QueryString;
+use ipl\Web\Url;
+
+class CreateHostSlaReport extends HostsDetailExtensionHook
+{
+    public function getHtmlForObjects(Query $hosts): ValidHtml
+    {
+        $url = Url::fromPath('reporting/reports/new');
+        $filter = QueryString::render($this->getBaseFilter());
+        $url->addParams(['filter' => $filter, 'report' => 'host']);
+        $wrapper = Html::tag('div');
```

Comment:
Please don't use a wrapper element. The other content also doesn't use one. Use a `HtmlDocument` as wrapper instead.